app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -118,6 +119,14 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     private var qrButtonAction: Toolbar.Action? = null
     private var voiceSearchButtonAction: Toolbar.Action? = null
 
+    private val cameraPermissionLauncher = registerCameraLauncher(
+        permissionGranted = { qrFeature.get()?.onPermissionsResult(arrayOf(CAMERA), intArrayOf(0)) },
+        permissionDenied = { forever: Boolean ->
+            if( forever ) interactor.onCameraPermissionsNeeded()
+            else resetFocus()
```

Comment:
Do not `resetFocus()` when we open the permission dialog to prevent reopening the keyboard while the dialog is open.